src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1154,6 +1163,109 @@ private static bool IsDynamicStatics(TypeDesc type)
+                // (see ShouldSkipCompilation) but get JITted instead. The JITted method will have the correct
+                // answer for the CPU the code is running on.
+                fTreatAsRegularMethodCall = (fIsGetIsSupportedMethod && fIsPlatformHWIntrinsic) || (!fIsPlatformHWIntrinsic && fIsHWIntrinsic);
+
+                if (_TARGET_X86_ || _TARGET_AMD64_)
+                {
+                    if (fIsPlatformHWIntrinsic)
+                    {
+                        // Simplify the comparison logic by grabbing the name of the ISA
+                        string isaName = (enclosingClassName == null) ? className : enclosingClassName;
+                        if ((isaName == "Sse") || (isaName == "Sse2"))
+                        {
+                            if ((enclosingClassName == null) || (className == "X64"))
+                            {
+                                // If it's anything related to Sse/Sse2, we can expand unconditionally since this is
+                                // a baseline requirement of CoreCLR.
+                                fTreatAsRegularMethodCall = false;
+                            }
+                        }
+                        else if ((className == "Avx") || (className == "Fma") || (className == "Avx2") || (className == "Bmi1") || (className == "Bmi2"))
+                        {
+                            if ((enclosingClassName == null) || (string.Equals(className, "X64")))
+                            {
+                                // If it is the get_IsSupported method for an ISA which requires the VEX
+                                // encoding we want to expand unconditionally. This will force those code
+                                // paths to be treated as dead code and dropped from the compilation.
+                                //
+                                // For all of the other intrinsics in an ISA which requires the VEX encoding
+                                // we need to treat them as regular method calls. This is done because RyuJIT
+                                // doesn't currently support emitting both VEX and non-VEX encoded instructions
+                                // for a single method.
+                                fTreatAsRegularMethodCall = !fIsGetIsSupportedMethod;
+                            }
+                        }
+                    }
+                    else if (namespaceName == "System")
+                    {
+                        if ((className == "Math") || (className == "MathF"))
+                        {
+                            // These are normally handled via the SSE4.1 instructions ROUNDSS/ROUNDSD.
+                            // However, we don't know the ISAs the target machine supports so we should
+                            // fallback to the method call implementation instead.
+                            fTreatAsRegularMethodCall = (methodName == "Round");
```

Comment:
Same for `Floor` and `Ceiling` ?

Reply:
Good point!

Those are handled at a different spot though. @cshung Looks like you missed porting the `getIntrinsicID` part of #25365.

Reply:
Thanks for the feedback - yes it is indeed missed.